feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go

```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
+	validateTrafficFlows(t, args.ate, args.top, BaseFlow, false)
 }
 
 // createFlow returns a flow from atePort1 to the dstPfx
-func createFlow(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology, name string) *ondatra.Flow {
-	srcEndPoint := top.Interfaces()[atePort1.Name]
-	dstEndPoint := []ondatra.Endpoint{}
-	for intf, intf_data := range top.Interfaces() {
-		if intf != "atePort1" {
-			dstEndPoint = append(dstEndPoint, intf_data)
-		}
-	}
-	hdr := ondatra.NewIPv4Header()
-	hdr.WithSrcAddress(dutPort1.IPv4).DstAddressRange().WithMin(dstPfxMin).WithCount(1)
-
-	flow := ate.Traffic().NewFlow(name).
-		WithSrcEndpoints(srcEndPoint).
-		WithDstEndpoints(dstEndPoint...).
-		WithHeaders(ondatra.NewEthernetHeader(), hdr).WithFrameSize(300)
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string) string {
+
+	flow := top.Flows().Add().SetName(name)
+	flow.Metrics().SetEnable(true)
+	e1 := flow.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	flow.TxRx().Port().SetTxName(atePort1.Name)
+	waitOTGARPEntry(t)
+	dstMac := ate.OTG().Telemetry().Interface(atePort1.Name + ".Eth").Ipv4Neighbor(dutPort1.IPv4).LinkLayerAddress().Get(t)
+	e1.Dst().SetChoice("value").SetValue(dstMac)
+	v4 := flow.Packet().Add().Ipv4()
+	v4.Src().SetValue(atePort1.IPv4)
+	v4.Dst().Increment().SetStart(dstPfxMin).SetCount(1)
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
+	return name
 
-	return flow
 }
 
 // validateTrafficFlows verifies that the flow on ATE and check interface counters on DUT
-func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, flow *ondatra.Flow, drop bool, d_port []string) {
-	ate.Traffic().Start(t, flow)
+func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, ateTop gosnappi.Config, flow string, drop bool) {
+	ate.OTG().StartTraffic(t)
 	time.Sleep(60 * time.Second)
-	ate.Traffic().Stop(t)
-	flowPath := ate.Telemetry().Flow(flow.Name())
+	ate.OTG().StopTraffic(t)
+	otgutils.LogFlowMetrics(t, ate.OTG(), ateTop)
+	otgutils.LogPortMetrics(t, ate.OTG(), ateTop)
+	flowPath := ate.OTG().Telemetry().Flow(flow)
 	got := flowPath.LossPct().Get(t)
 	if drop {
 		if got != 100 {
-			t.Fatalf("Traffic passing for flow %s got %f, want 100 percent loss", flow.Name(), got)
+			t.Fatalf("Traffic passing for flow %s got %f, want 100 percent loss", flow, got)
 		}
 	} else {
 		if got > 0 {
-			t.Fatalf("LossPct for flow %s got %f, want 0", flow.Name(), got)
+			t.Fatalf("LossPct for flow %s got %f, want 0", flow, got)
 		}
 	}
 }
 
 // flapinterface shut/unshut interface, action true bringsup the interface and false brings it down
-func flapinterface(t *testing.T, ate *ondatra.ATEDevice, port string, action bool) {
-	ateP := ate.Port(t, port)
-	ate.Actions().NewSetPortState().WithPort(ateP).WithEnabled(action).Send(t)
+func flapinterface(t *testing.T, dut *ondatra.DUTDevice, port string, action bool) {
+	// When running on kne the corresponding DUT port will be taken down
```

Comment:
I would rephrase this/also mention that trying to shut OTG ports in KNE has no effect.

Reply:
Done